extension/grpcindexer/external_subscriber_client.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package grpcindexer
+
+import (
+	"context"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/extension/indexer"
+
+	pb "github.com/ava-labs/hypersdk/proto"
+)
+
+var _ indexer.AcceptedSubscriber = (*ExternalSubscriberClient)(nil)
+
+type ExternalSubscriberClient struct {
+	client pb.ExternalSubscriberClient
+}
+
+func NewExternalSubscriberClient(ctx context.Context, server string, networkID uint32, chainID ids.ID, genBytes []byte) (*ExternalSubscriberClient, error) {
```

Comment:
Could we rename `server -> serverAddr` and `genBytes -> genesisBytes` ? With this number of arguments, can we put one argument on each line and the return value on the final line as `) (*ExternalSubscriberClient, error) {`